R/run_pacta_reporting_process.R

```diff
@@ -0,0 +1,317 @@
+run_pacta_reporting_process <- function(
+  raw_params = commandArgs(trailingOnly = TRUE),
+  analysis_output_dir = Sys.getenv("ANALYSIS_OUTPUT_DIR"),
+  benchmarks_dir = Sys.getenv("BENCHMARKS_DIR"),
+  report_output_dir = Sys.getenv("REPORT_OUTPUT_DIR"),
+  summary_output_dir = Sys.getenv("SUMMARY_OUTPUT_DIR"),
+  real_estate_dir = Sys.getenv("REAL_ESTATE_DIR"),
+  survey_dir = Sys.getenv("SURVEY_DIR"),
+  score_card_dir = Sys.getenv("SCORE_CARD_DIR")
+) {
+  log_debug("Checking configuration.")
+  if (is.null(analysis_output_dir) || analysis_output_dir == "") {
+    log_error("ANALYSIS_OUTPUTS_DIR not set.")
+    stop("ANALYSIS_OUTPUTS_DIR not set.")
+  }
+  if (is.null(benchmarks_dir) || benchmarks_dir == "") {
+    log_error("BENCHMARKS_DIR not set.")
+    stop("BENCHMARKS_DIR not set.")
+  }
+  if (is.null(report_output_dir) || report_output_dir == "") {
+    log_error("REPORT_OUTPUT_DIR not set.")
+    stop("REPORT_OUTPUT_DIR not set.")
+  }
+  if (is.null(summary_output_dir) || summary_output_dir == "") {
+    log_error("SUMMARY_OUTPUT_DIR not set.")
+    stop("SUMMARY_OUTPUT_DIR not set.")
+  }
+
+  # defaulting to WARN to maintain current (silent) behavior.
+  logger::log_threshold(Sys.getenv("LOG_LEVEL", "INFO"))
+  logger::log_formatter(logger::formatter_glue)
+
+  # -------------------------------------------------------------------------
+
+  log_info("Starting portfolio report process")
+
+  # Read Params
+  log_trace("Processing input parameters.")
+  if (length(raw_params) == 0L || all(raw_params == "")) {
+    log_error("No parameters specified.")
+  }
+
+  # log_trace("Validating raw input parameters.")
+  # raw_input_validation_results <- jsonvalidate::json_validate(
+  #   json = raw_params,
+  #   schema = system.file(
+  #     "extdata", "schema", "rawParameters.json",
+  #     package = "workflow.pacta.report"
+  #   ),
+  #   verbose = TRUE,
+  #   greedy = FALSE,
+  #   engine = "ajv"
+  # )
+  # if (raw_input_validation_results) {
+  #   log_trace("Raw input parameters are valid.")
+  # } else {
+  #   log_error(
+  #     "Invalid raw input parameters. ",
+  #     "Must include \"inherit\" key, or match full schema."
+  #   )
+  #   stop("Invalid raw input parameters.")
+  # }
```

Comment:
Should this be removed?

Reply:
I don't think it needs to be removed, since it will be enabled soon. This bit handles the validation, once the JSON Schema is defined.